collect_app/src/main/java/org/odk/collect/android/adapters/HierarchyListAdapter.java

```diff
@@ -48,17 +50,24 @@ public HierarchyListAdapter(List<HierarchyElement> listElements, OnElementClickL
 
     @Override
     public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
-        holder.bind(hierarchyElements.get(position), listener);
-        if (hierarchyElements.get(position).getIcon() != null) {
+        HierarchyElement element = hierarchyElements.get(position);
+        holder.bind(element, listener);
+        Drawable icon = element.getIcon();
+        if (icon != null) {
             holder.icon.setVisibility(View.VISIBLE);
-            holder.icon.setImageDrawable(hierarchyElements.get(position).getIcon());
+            holder.icon.setImageDrawable(icon);
         } else {
             holder.icon.setVisibility(View.GONE);
         }
-        holder.primaryText.setText(HtmlUtils.textToHtml(hierarchyElements.get(position).getPrimaryText()));
-        if (hierarchyElements.get(position).getSecondaryText() != null && !hierarchyElements.get(position).getSecondaryText().isEmpty()) {
+        holder.primaryText.setText(HtmlUtils.textToHtml(element.getPrimaryText()));
+        String secondaryText = element.getSecondaryText();
+        if (secondaryText != null && !secondaryText.isEmpty()) {
             holder.secondaryText.setVisibility(View.VISIBLE);
-            holder.secondaryText.setText(HtmlUtils.textToHtml(hierarchyElements.get(position).getSecondaryText()));
+            //For #4480
+            holder.secondaryText.setText(
+                    element.getType() == HierarchyElement.Type.QUESTION
```

Comment:
Nice catch! I'm wondering if this can be even simpler however: is there ever a case where we would set HTML as `secondaryText`? It looks like in all cases other than `QUESTION` it gets set to a static label for the type of element (like "Repeat" or "Group").

Reply:
Indeed, do please simplify further if you're certain it's safe.

Reply:
I've changed it so it never uses HTML rendering for the secondary text. Would be good for @grzesiek2010 or @lognaturel to review now as I've touched it too much to be unbiased I reckon!

Reply:
Now such a neat fix that it scarcely needs the test?

Reply:
You could argue that. I'd want a test for even a small change like this to enforce that the behaviour we want stays there. In a big ol' code base like this it's also advantageous to have a test that documents that something is intentional and not just there because of duplication. I had to double-check we don't want HTML rendering there, but now it's explicit. You'll see that writing the test also pushed me to extract a separate view class which, in my opinion, is an improvement to the overall structure here - it keeps the adapter purely boilerplate.

I guess an interesting point is that, if I was writing this view class from scratch, I would probably have written a simpler test that just checks that the string is displayed and wouldn't have bothered with a case for HTML as I'd have assumed "no one is going to bother adding the HTML rendering if it's not needed".

Reply:
Looks good I added one commit too, why not :) (to improve the new test).